schema/auth.json

```diff
@@ -0,0 +1,214 @@
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "type": "string",
+              "description": "Expression referencing a workflow secret that contains all needed auth info"
+            },
+            {
+              "title": "Basic Auth Info",
+              "$ref": "#/definitions/basicinfodef"
+            },
+            {
+              "title": "Bearer Auth Info State",
+              "$ref": "#/definitions/bearerinfodef"
+            },
+            {
+              "title": "Oidc Auth Info",
+              "$ref": "#/definitions/oidcinfodef"
+            }
+          ]
+        }
+      },
+      "required": [
+        "name",
+        "info",
+        "scheme"
+      ]
+    },
+    "basicinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed basic auth info"
+        },
+        {
+          "type": "object",
+          "description": "Basic auth information",
+          "properties": {
+            "username": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user name",
+              "minLength": 1
+            },
+            "password": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the user password",
+              "minLength": 1
+            },
+            "metadata": {
+              "$ref": "common.json#/definitions/metadata"
+            }
+          },
+          "required": [
+            "username",
+            "password"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "bearerinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed bearer auth info"
+        },
+        {
+          "type": "object",
+          "description": "Bearer auth information",
+          "properties": {
+            "token": {
+              "type": "string",
+              "description": "String or a workflow expression. Contains the token",
+              "minLength": 1
+            },
+            "metadata": {
+              "$ref": "common.json#/definitions/metadata"
+            }
+          },
+          "required": [
+            "token"
+          ],
+          "additionalProperties": false
+        }
+      ]
+    },
+    "oidcinfodef": {
+      "oneOf": [
+        {
+          "type": "string",
+          "description": "Expression referencing a workflow secret that contains all needed oidc auth info"
+        },
+        {
+          "type": "object",
+          "description": "Oidc auth information",
+          "properties": {
+            "grantType": {
+              "type": "string",
+              "description": "Defines the auth type",
+              "enum": [
+                "resourceOwner",
```

Comment:
Even though I don't like naming it that way, I think we probably should change that to `password`, which is the actual value you have to input when requesting a token with the **Resource Owner Password Credentials** grant type.

Reply:
hmm i am not expert in this but this enum values look confusing to me :) 

             "enum": [
                "password",
                "clientCredentials",
                "tokenExchange"
              ],

is this ok ?

Reply:
updated

Reply:
@tsurdilo Yes, thanks!!!